packages/@aws-cdk/core/lib/duration.ts

```diff
@@ -9,6 +9,11 @@ import { Token } from "./token";
  * Whent he amount is passed as an token, unit conversion is not possible.
  */
 export class Duration {
+  /**
+   * A reference `now` that can be used throughout the CDK.
+   */
+  public static now = Date.now();
```

Comment:
I wholeheartedly agree with the introduction of this member. Just not sure whether it should be on `Duration`.

Reply:
I've been thinking and I can't come up with a good name. Romain prefers `Timestamp.now` which feels too mechanical for my tastes (not what we're trying to describe). I prefer `Moment.now` but Romain thinks that word will confuse people w.r.t. the "moment" library.

To avoid having to settle the debate here and now, can you just make this field a private variable of the S3 deployment module? It seems like the S3 deployment module is the only one that uses `Date.now` so far.

Reply:
Got it 😄 _There are only two hard things in Computer Science..._